### PR TITLE
Pass context to various components

### PR DESCRIPTION
**Pull request by eaudetcobello**
Searched for Enable\w+Component, Disable\w+Component.

### CODE REVIEW

File: src/k8s/pkg/component/dns.go [hunk 1]
```diff
@@ -4,13 +4,12 @@ import (
...
 	"fmt"
 	"strings"
-	"time"
 
 	"github.com/canonical/k8s/pkg/snap"
 	"github.com/canonical/k8s/pkg/utils/k8s"
 )
 
-func EnableDNSComponent(s snap.Snap, clusterDomain, serviceIP string, upstreamNameservers []string) error {
+func EnableDNSComponent(s snap.Snap, clusterDomain, serviceIP string, upstreamNameservers []string, ctx context.Context) error {
```
Comment (neoaggelos) [inline src/k8s/pkg/component/dns.go:12]: it's idiomatic that the context be the first argument on all functions

Comment (neoaggelos) [reply src/k8s/pkg/component/dns.go:12]: also update all other places (and caller sites).

Comment (neoaggelos) [inline src/k8s/pkg/component/dns.go:12]:
```suggestion
func EnableDNSComponent(ctx context.Context, s snap.Snap, clusterDomain, serviceIP string, upstreamNameservers []string) error {
```

**Review by neoaggelos — COMMENTED** [review]
Great work, let's make sure that we write idiomatic Go code and we're good to go!

**Review by neoaggelos — COMMENTED** [review]
Needs rebase.

File: src/k8s/pkg/k8sd/api/component.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ func putDNSComponent(s *state.State, r *http.Request) response.Response {
 
 	switch req.Status {
 	case api.ComponentEnable:
-		dnsIP, clusterDomain, err := component.EnableDNSComponent(snap, req.Config.ClusterDomain, req.Config.ServiceIP, req.Config.UpstreamNameservers)
+		dnsIP, clusterDomain, err := component.EnableDNSComponent(s.Context, snap, req.Config.ClusterDomain, req.Config.ServiceIP, req.Config.UpstreamNameservers)
```
Comment (bschimke95) [inline src/k8s/pkg/k8sd/api/component.go:46]: We should use the request context here and below: `r.Context`
This request context is cancelled if the user terminates the request. The state context is only cancelled if `k8sd` terminates.

**Review by bschimke95 — COMMENTED** [review]
Great work @eaudetcobello !
Only a minor change and we are good to go.

**Review by bschimke95 — APPROVED** [review]
LGTM, great work!

**Pull request merged**